src/basekernels/piecewisepolynomial.jl

```diff
@@ -10,12 +10,14 @@ processes are hence v times  mean-square differentiable. The kernel function is:
 where `r` is the Mahalanobis distance mahalanobis(x,y) with `maha` as the metric.
 
 """
-struct PiecewisePolynomialKernel{V, A<:AbstractMatrix{<:Real}} <: BaseKernel
+struct PiecewisePolynomialKernel{V, A<:AbstractMatrix{<:Real}} <: SimpleKernel
     maha::A
+    j::Int64
```

Comment:
```suggestion
    j::Int
```
I'm also wondering if one could come up with a more descriptive name for `j`?

Reply:
Could we please leave this for a separate PR? This is already quite large.

Reply:
IMO we should at least make PiecewisePolynomialKernel a SimpleKernel (or whatever it is called), but all other optimizations might indeed be better to move to a separate PR.

Reply:
Agreed.

Reply:
The `Int64` -> `Int` is just formatting right?
For the name, I saw the denomination `extent` but it's not really helpful. Since it is an intermediary computed quantity it's probably better to leave it as it is.

Reply:
No, that's not just formatting.

Reply:
To expand, whether or not `Int` is `Int64` or `Int32` depends whether your architecture is 32 bit or 64 bit. See [here](https://docs.julialang.org/en/v1/manual/integers-and-floating-point-numbers/#Integers-1) :)